jina/drivers/__init__.py

```diff
@@ -303,14 +307,20 @@ def pre_traverse(_docs, recur_on, context_doc=None):
         else:
             raise ValueError(f'{self.recursion_order}')
 
-        if 'chunks' in self.traverse_fields:
-            depth_name = 'granularity'
-            _traverse(docs, 'chunks')
+        if (self._depth_start < self._depth_end) or \
+                (self._depth_start == self._depth_end and self._depth_start > 0):
```

Comment:
Currently we are catching the `_depth_start=0` case on `apply` and `apply_all` level (at least for the ranker). I like, preventing it beforehand. Would it be a good idea, to remove the `if context_doc is None: return` logic from the driver or is this it better to catch twice?

Reply:
I honestly was not thinking about this when doing this, here I was mostly focused on translating really explicitly the old content. I would like to remove but I would prefer to have it in another PR if possible

Reply:
I agree with Joan, the changes on the logic need more test. Let's make another PR for that.